### ISSUE TITLE
Sync/ Device discovery inconsistent

### ISSUE THREAD

**Issue by ximenabb** [digidem/comapeo-core#872]
Details : https://www.notion.so/digidem/Sync-Device-discovery-inconsistent-10c1b08162d580099a26fdeb94138c5c

**Comment by EvanHahn** [comment]
Andrew was still able to reproduce this. I was hopeful that we'd fixed it, but alas.

**Comment by ximenabb** [comment]
Update - https://www.notion.so/digidem/Multi-Project-Discovery-issue-20e1b08162d58025bb2af5a3b83933af

To review in Sentry

**Comment by RangerMauve** [comment]
Yeah the main issue is that our message channel in nodejs-mobile is not an actual message channel. It doesn't queue items until start is called because there's a gap between where it's initialized and where the node thread starts at all.

https://github.com/digidem/comapeo-mobile/blob/develop/src/backend/src/message-port-like.js#L41

I think what we really need to do is wait on the front end until we get an initial message from the nodejs side over the bridge. Then we could leave IPC as is. Though updating IPC could be good to future proof for this scenario.

Also due to how IPC is structured, it might not give the right hooks to wait for messages to go across. Ideally I'd want to have `createMapeoClient` be async or to have a way for the proxy to queue up deferred promises inside the [get](https://github.com/digidem/comapeo-ipc/blob/main/src/client.js#L47).

There's no obvious point to listen on in the rpc-reflector side either since it's immediately trying to proxy messages over. https://github.com/digidem/rpc-reflector/blob/main/lib/client.js

**Comment by RangerMauve** [comment]
With the latest PR from gregor this should be fixed unless there is some other issue we haven't found yet. 👍

Will close this after the next round of QA

**Comment by RangerMauve** [comment]
Added another fix to reuse the port when local discovery server gets started/stopped/restarted.

Also added a bunch more logs to help catch invite errors
